### DISCUSSION TITLE
Xiao esp32 and tft_espi - c3 works well/ s3 doesn't work.. What are the differences (Solved)

### DISCUSSION THREAD

**Discussion by bamoore01** [Bodmer/TFT_eSPI#2757] [Compatible displays and setup files]
I have a Seeed Xiao esp32c3 and an esp32S3 board that I am using with a Waveshare 2"LCD Module (ST7789).  I can get the c3 board to work well with the LCD using the tft_espi drivers but I can not figure out what I'm doing wrong with the s3 board.  All I get from the s3 board is a blank screen.

It appears that both the c3 and s3 boards have the same pinout as far as the SPI configuration is concerned so I am simply swapping between the two boards on a breadboard with no wiring or configuration changes.  Everything in the IDE is up to date on the latest versions.  I do switch between the c3 and s3 in the IDE when I'm changing the boards.  I simplified the sketch to as simple as I could to try to make this work (below).  The commented section at the end is the relevant part of the user_setup.h from the tft_espi library directory.  With this sketch, the LCD will change color to whatever I tell it to be with the fillScreen command for the c3 but when I use the se board, I just get a blank, black screen.

Any help understanding what I'm doing wrong would help.
Thanks 

#include <SPI.h>
#include <TFT_eSPI.h>       // Hardware-specific library
TFT_eSPI    tft  = TFT_eSPI();  // Invoke custom library
void setup() {
  tft.begin();
  tft.fillScreen(TFT_BLUE);
  }

void loop() {
  // do nothing
  }

//#define TFT_MISO -1
//#define TFT_MOSI D10
//#define TFT_SCLK D8
//#define TFT_CS   D7  // Chip select control pin
//#define TFT_DC   D3  // Data Command control pin
//#define TFT_RST  D4  // Reset pin (could connect to RST pin)

**Comment by Bodmer** [comment]
Use the default SPI pins as noted here:

#define TFT_CS   34 //     10 or 34 (FSPI CS0) 
#define TFT_MOSI 35 //     11 or 35 (FSPI D)
#define TFT_SCLK 36 //     12 or 36 (FSPI CLK)
#define TFT_MISO 37 //     13 or 37 (FSPI Q)

If you use GPIO 10-13 then add the following line to the setup:
#define USE_HSPI_PORT

SPI GPIO should not be randomly assigned for the S3. If other pins are used then it may be necessary to reduce the SPI clock rate due to the use of an IO multiplexer within the S3. Example pins for another board here:
https://github.com/Bodmer/TFT_eSPI/blob/master/User_Setups/Setup251_ESP32_S3_Box.h

**Comment by bamoore01** [comment]
Bodmer, Thanks for the info.  The Xiao doesn't have GPIO 34-37 or 10-13 available on the board but the info that you provided did get me on the right track.  It turns out that my real problem was simply that TFT_CS don't work on D7.  One of the examples you referenced (Xiao with the round display) used D1 for TFT_CS.  I changed it to D1 and the display lit up like it should (kinda).  I made a few other minor changes but the TFT_CS change is the main thing.  I'm posting a small sketch below that will work on this board/display combination.

The kinda add above is because of a display problem I'm still having but I'll post that in another thread.  (not sure why this is in bold??)
```
#include <SPI.h>
#include <TFT_eSPI.h> // Hardware-specific library

#define ST7789_2_DRIVER 
#define TFT_RGB_ORDER TFT_RGB  // Colour order Red-Green-Blue
#define TFT_INVERSION_ON

#define TFT_MISO -1
#define TFT_MOSI D10
#define TFT_SCLK D8
#define TFT_CS   D1  // Chip select control pin. Did not work on D7
#define TFT_DC   D3  // Data Command control pin
#define TFT_RST  D4  // Reset pin
#define TFT_BL   D6  

//#define SPI_FREQUENCY  27000000   //  It worked at 40000000 but I left this in incase something goes wrong
#define SPI_FREQUENCY  40000000

#define TFT_WIDTH  320
#define TFT_HEIGHT 240

TFT_eSPI tft = TFT_eSPI(); // Invoke custom library

void setup() {
  tft.begin();
  tft.fillScreen(TFT_RED);
  tft.setCursor(10,10);
  tft.setTextColor(TFT_WHITE, TFT_BLACK);
  tft.println("Hello "+String(TFT_WIDTH)+" "+String(TFT_HEIGHT));
}

void loop() {
  // do nothing
}
```

**Reply by adlernunez** [reply]
i have similar problem with ili9341 tft display! works well with esp32 nodemcu32s.
but not able to use with xiao esp32s3. i changed the pin no.s to respective pins on xiao. still not working.

i changed user setups to  setup66_seeed_xiao_round.h but its for a different display i.e GC9A01.

can anybody help me with step by step process.

**Reply by 2ant** [reply]
If you have a user setup that work with your screen, don't use an already made user setup for another screen, make your own.

-copy it into a new file
-make sure you use that new file in user_setup_select.h
-find a pin configuration that works with your board

you need to use the read_user_setup example sketch to make sure that you the library use the correct file.

i hadn't tested my xiao s3 yet so i did those steps and tested the following pins after @bamoore01 post and it works for a ST7735 screen.

```


#define USER_SETUP_ID 70

#define ST7735_DRIVER


#define TFT_WIDTH  128
#define TFT_HEIGHT 160

#define ST7735_GREENTAB2




#define TFT_CS   2      //D1  
#define TFT_DC    4     //D3    

#define TFT_MOSI 9      //D10   
#define TFT_MISO -1  
#define TFT_SCLK 7      //D8    


#define TFT_RST   5     //D4    


#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT

//#define SPI_FREQUENCY  27000000
#define SPI_FREQUENCY  40000000   // Maximum for ILI9341
//#define SPI_FREQUENCY  80000000   //corruption


```

**Comment by summerfind** [comment]
Hi Bodmer,
I used ESP32S3 + ST7789V2 1.69 inch and TFT_eSPI.h show the clock in not full screen, and abnormal color.
used the config: If you use GPIO 10-13 then add the following line to the setup:
#define USE_HSPI_PORT
what can be ?
